Project.java

```diff
@@ -498,17 +511,18 @@ public void run() {
     
 }
 
-public class Project3 {
+public class Project {
     public static void main(String[] args){
-        
+        FoodList.initiateFood();
         try
         {           
         File f = new File("backup");
         if(f.exists())
         {
             FileInputStream fin=new FileInputStream(f);
             ObjectInputStream ois=new ObjectInputStream(fin);
-            Hotel.ob=(holder)ois.readObject();
+            Hotel.hotel_ob=(holder) ois.readObject();
+            ois.close();
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/shouryaj98/Hotel-Management-Project-Java/pull/4#issuecomment-2437657272>#2aa38a</a><span> - 10/25/2024, 12:36 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Closing the ObjectInputStream without closing the underlying FileInputStream can lead to resource leaks.
</div>
<div id="fix">
<b>Fix:</b> Close both the ObjectInputStream and FileInputStream using try-with-resources or in a finally block to ensure proper resource management.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -522,6 +522,7 @@
      {
          FileInputStream fin=new FileInputStream(f);
          ObjectInputStream ois=new ObjectInputStream(fin);
 -        Hotel.hotel_ob=(holder) ois.readObject();
 -        ois.close();
 +        try (fin; ois) {
 +            Hotel.hotel_ob=(holder) ois.readObject();
 +        }
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged